### DISCUSSION TITLE
Expanding a `Frame` to panel size

### DISCUSSION THREAD

**Discussion by aleokdev** [emilk/egui#1447] [Q&A] [answered]
How can I expand a `Frame` to include the contents of the entire panel it is in, without any margin/borders/etc?

**Comment by emilk** [comment, marked as answer]
I don't understand the question.

If you're asking how to use up the rest of the available space, the answer is to put `ui.allocate_space(ui.available_size());` as the last thing inside the `Frame`.

**Reply by aleokdev** [reply]
That does use up the rest of space, but keeps the frame's margin intact, even with `.margin(Margin{left: 0., /*...*/})`. What I'd like to have is like plots: Only the background color, no spacing between frame background and panel limits. Am I using the wrong tool? I'd just like a canvas, nothing else.
![image](https://user-images.githubusercontent.com/16354904/161440821-781a61e5-ebb3-49fa-9d51-b9e75b40b980.png)

**Reply by emilk** [reply]
Oh, you want to use `CentralPanel::frame`

**Reply by aleokdev** [reply]
Thank you! That is exactly what I wanted.
